packages/flutter/lib/src/widgets/dismissible.dart

```diff
@@ -232,10 +238,25 @@ class Dismissible extends StatefulWidget {
   /// This defaults to [HitTestBehavior.opaque].
   final HitTestBehavior behavior;
 
+  /// Called when the dismissible widget has been dragged.
+  final DismissUpdateCallback? onDismissUpdate;
+
   @override
   State<Dismissible> createState() => _DismissibleState();
 }
 
+/// Contains various values related to the drag progress of a Dismissible
+class DismissUpdate {
+  /// Create a new instance
+  DismissUpdate({this.reached = false, this.previousReached = false});
+
+  /// Whether the DismissThreshold is currently reached
+  bool reached;
+
+  /// Whether the DismissThreshold was previously reached
```

Comment:
```suggestion
  /// Whether the dismiss threshold was reached the last time this callback was invoked.
  ///
  /// This can be used in conjunction with [DismissUpdate.reached] to catch the moment
  /// that the [Dismissible] is dragged across the threshold.
```